src/engine/guess_toolset.bat

```diff
@@ -32,6 +32,21 @@ if NOT "_%VS150COMNTOOLS%_" == "__" (
     set "BOOST_JAM_TOOLSET=vc141"
     set "BOOST_JAM_TOOLSET_ROOT=%VS150COMNTOOLS%..\..\VC\"
     goto :eof)
+
+set VSWHERE_PRE=Microsoft.VisualStudio.Product.
+set VSWHERE_IDS=%VSWHERE_PRE%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+set VSWHERE_REQ=-requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
+set VSWHERE_PRP=-property installationPath
+set VSWHERE="%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% set VSWHERE="%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% goto :skip_vswhere
+for /f "usebackq tokens=*" %%i in (`%VSWHERE% -latest -products %VSWHERE_IDS% %VSWHERE_REQ% %VSWHERE_PRP%`) do (
```

Comment:
> ASAIK -products * does not cover "Buildtools" ;( (Microsoft/vswhere#8) but I'll check.

Without `-products` does not cover "Buildtools", but `-products *` cover all products https://github.com/Microsoft/vswhere/issues/23

> -latest is supposed to be kinda similar to -version "[15.0,16.0)", no?

"-latest" will find even theoretical "Visual Studio 2018/2019" (> 16.0). I think we should limit to VS2017 only